### PR TITLE
Sensirion sdp ver2

### PR DESCRIPTION
**Pull request by JonBruchim**

### CODE REVIEW

File: drivers/sdp8_pressure_sensor.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        log.info("SDP pressure sensor initialized")
+
+    def _start_measure(self):
+        try:
+            self._pig.i2c_write_device(self._dev, self.START_MEASURE_FLOW_AVG_CMD)
+        except pigpio.error as e:
+            log.error("Could not write start_measure cmd to flow sensor. "
+                      "Is the flow sensor connected?.")
+            raise I2CWriteError("i2c write failed")
+        sleep(0.1)
```
Comment (osherdp) [inline drivers/sdp8_pressure_sensor.py:61]: why?

File: drivers/sdp8_pressure_sensor.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    def _pressure_to_flow(self, pressure):
+        flow = (abs(pressure) ** 0.5) * self.SYSTEM_RATIO
+        if pressure < 0:
+            flow = -flow
+
+        return flow
+
+    def twos_complement(self, number):
+        import sys
```
Comment (osherdp) [inline drivers/sdp8_pressure_sensor.py:80]: import it at top

File: drivers/dsp_test.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from sdp8_pressure_sensor import SdpPressureSensor
```
Comment (osherdp) [inline drivers/dsp_test.py:1]: in the file's name you mean `sdp_test.py`?
also move to  `drivers/debug/print_sdp8.py`

File: drivers/errors.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+"""Errors module."""
```
Comment (osherdp) [inline drivers/errors.py]: This is a duplicated module

File: drivers/sdp8_pressure_sensor.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        sleep(0.1)
+
+        log.info("Started flow sensor measurement")
+
+    def _calculate_pressure(self, pressure_reading):
+        differential_psi_pressure =\
+            pressure_reading / (self.SCALE_FACTOR_PASCAL)
+        differential_cmh2o_pressure =\
+            differential_psi_pressure * (1 / self.CMH20_PASCAL_RATIO)
+        return (differential_cmh2o_pressure)
```
Comment (osherdp) [inline drivers/sdp8_pressure_sensor.py:70]: remove redundant parenthesis

File: drivers/sdp8_pressure_sensor.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    def read(self):
+        """ Returns pressure as flow """
+        try:
+            read_size, pressure_raw =\
+                self._pig.i2c_read_device(self._dev, self.MEASURE_BYTE_COUNT)
+
+            if read_size >= self.MEASURE_BYTE_COUNT:
+                pressure_reading = (pressure_raw[0] << 8) | (pressure_raw[1])
+                pressure_reading = self.twos_complement(pressure_reading)
+                expected_crc = 0#pressure_raw[2]
```
Comment (osherdp) [inline drivers/sdp8_pressure_sensor.py:106]: fix as we talked

File: drivers/sdp8_pressure_sensor.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                crc_calc = expected_crc #self._crc8(pressure_reading)
+                if not crc_calc == expected_crc:
+                    print('bad crc')
+                return (self._pressure_to_flow(self._calculate_pressure(pressure_reading)))
+            else:
+                log.error("Pressure sensor's measure data not ready")
+                raise I2CReadError("Pressure sensor measurement unavailable.")
+        except pigpio.error as e:
+            log.error("Could not read from pressure sensor. "
+                      "Is the pressure sensor connected?.")
```
Comment (osherdp) [inline drivers/sdp8_pressure_sensor.py:116]: delete the '.'

File: drivers/errors.py (hunk 4 above)
Comment (JonBruchim) [reply drivers/errors.py]: removed this

File: drivers/sdp8_pressure_sensor.py (hunk 1 above)
Comment (JonBruchim) [reply drivers/sdp8_pressure_sensor.py:61]: Reznic didn't want the measurments to happen directly after the init
Guess it's not relevant

Comment (osherdp) [reply drivers/sdp8_pressure_sensor.py:63]: then delete

**Pull request merged**